homeassistant/components/media_player/yamaha.py

```diff
@@ -0,0 +1,96 @@
+"""
+Support for Yamaha Receivers.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/media_player.yamaha/
+"""
+import logging
+
+from homeassistant.components.media_player import (
+    SUPPORT_TURN_OFF, SUPPORT_TURN_ON, SUPPORT_VOLUME_MUTE, SUPPORT_VOLUME_SET,
+    MediaPlayerDevice)
+from homeassistant.const import STATE_OFF, STATE_ON
+REQUIREMENTS = ['rxv==0.1.9']
+_LOGGER = logging.getLogger(__name__)
+
+SUPPORT_YAMAHA = SUPPORT_VOLUME_SET | SUPPORT_VOLUME_MUTE | \
+    SUPPORT_TURN_ON | SUPPORT_TURN_OFF
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Yamaha platform."""
+    yamaha = YamahaDevice(config.get("name", "Yamaha Receiver"))
+    if yamaha.update():
+        add_devices([yamaha])
+        return True
+    else:
+        return False
+
+
+class YamahaDevice(MediaPlayerDevice):
+    """Representation of a Yamaha device."""
+
+    # pylint: disable=too-many-public-methods, abstract-method
+    def __init__(self, name):
+        """Initialize the Yamaha Receiver."""
+        import rxv
+        self._receiver = rxv.find()[0]
+        self._muted = False
+        self._volume = 0
+        self._pwstate = STATE_OFF
+        self.update()
+        self._name = name
+
+    def update(self):
+        """Get the latest details from the device."""
+        if self._receiver.on:
+            self._pwstate = STATE_ON
+        else:
+            self._pwstate = STATE_OFF
+        self._muted = self._receiver.mute
+        self._volume = (self._receiver.volume/100) + 1
+        return True
```

Comment:
Sorry, I cant see what line you commented on.  Maybe im just not looking in the right spot.  However, I did refactor this.